### ISSUE TITLE
Student Enrollment Trends

### ISSUE THREAD

**Issue by NatalieRaketic** [hackoregon/civic#197]
- [x] Needs API details

# Story Card Request

School Map and School Enrollment and Race/Ethnicity Demographics Over Time

## Description / Wireframe

Map of Portland Public schools and  2 Stacked Area graph of subpopulation changes over time. Map will be interactive, and when a school is selected, 2 stacked area graphs will populate. One will show counts of white students, asian students and historically underrepresented students (aggregation of all other race/ethnicities). The second will be a break out stacked area graph showing which details the subpopulations of the historically underrrepresented race/ethnicity groups (African American/Black, Hispanic/Latino, Native American/American Indian and Native Hawaiian/Pacific Islander). When no school is selected, all of PPS in aggregate is shown. 

On Map: School type will be color coded (5 categories : PK-5,PK-8/PK-12,6-8,9-12). Icon size will be based on total enrollment (though some schools are quite small, so a minimum size needs to be set so all school are clearly visible).

## Data Visualization / Design Concerns

![img_9380](https://user-images.githubusercontent.com/23616408/41205040-90451e82-6ca1-11e8-850e-c23c8d4d3ae5.jpg)

![img_9332](https://user-images.githubusercontent.com/23616408/41205051-a6bc0f18-6ca1-11e8-921e-10a8b56dd9b5.jpg)

![img_9381](https://user-images.githubusercontent.com/23616408/41205444-62e0596e-6ca8-11e8-88a7-4fe9bee1d6d0.jpg)

## Context
> Why is this important? This information will be used to prioritize and contextualize story cards.

## Data Details (APIs)
ode_enrollment.csv 

Columns: 

American Indian/Alaska Native, Asian, Native Hawaiian/Pacific Islander, Black/African American	Hispanic/Latino, White, Multi-Racial, year, Total enrollment

We need to calculate: Total enrollment minus (white + asian) = Historically underrepresented Population ( this becomes its own column) @jethard (please edit if not correct)

### See the source
>Link to API / data documentation on GitHub
>Link to original source data

**Context**

**Data Details (APIs)**

Datasets: 
Enrollment by Race-Ethnicity

**Comment by jaronheard** [comment]
More detail about race-ethnicity categories:

We've designed to support for 5 categories, but there are 7 categories on this chart. One possibility is showing the categories that PPS uses to make decisions, which is:
Historically advantaged: White, Asian, and White/Asian
Historically disadvantaged: All other categories
I'm not saying it's a good way to categorize, but that's what PPS is doing to make decisions and that's worth calling out.

**Comment by hassanshamim** [comment]
There are a few rows in ode_enrollment that have non-numeric data in the 'Total Enrollment' column.    i.e.: "Merged with Damascus Middle School ID 90 to become Deep Creek-Damascus K-8 School ID 90" or "Closed"

Should I discard these rows or keep them in the data and just not do anything useful man with this column?

**Comment by hassanshamim** [comment]
API responses for:
 - total portland race breakdown (by year)
 - by school (still aggregated statistics, filter by school name)
 - all school names
- give them the student percentage for the racial breakdowns.
- 'Historically Under represented minorities'  will be summed up on the frontend, they include all the races that aren't 'white' or 'asian'.

so example API response:

**Comment by hassanshamim** [comment]
So getting the schools geocoded by our newest (and best!) dataset doesn't seem feasible in the timeframe remaining.  I hope a simple dropdown without the map will still be feasible.

New endpoints for this card:
neighborhood-development/api/school_demographics/names - To get the unique list of school names

the `school_demographics` endpoint can be filtered by school name

lastly, `school_demographics/yearly_totals` will give you the total # of students, as well as the calculated amount of each racial type by year.

**Comment by hassanshamim** [comment]
so the api links mentioned above:
school names: http://service.civicpdx.org/neighborhood-development/api/school_demographics/names

filtering by school name: http://service.civicpdx.org/neighborhood-development/api/school_demographics?year=&type=&name=Beverly+Cleary

yearly breakdown: http://service.civicpdx.org/neighborhood-development/api/school_demographics/yearly_totals?limit=13

**Comment by hassanshamim** [comment]
@jaronheard ping me if there's anything I missed :)

**Comment by jaronheard** [comment]
@hassanshamim endpoints look like they're the right shape! 🙌

@hassanshamim @arailsya there are null values and zero values in the API response - does null represent something different than 0? not sure this is a problem, but might be better to return 0 values where there is supposed to be a number.

@arailsya - we're back to 8 categories, but we can just show unspecified with chart 1. Can you indicate what labels to show for what categories from the API response?
```
chart 1:
            "enroll_white": 78.8,
            "enroll_asian": 3.0,
            "enroll_unspecified": null,
HISTORICALLY UNDERREPRESENTED = sum of all categories below

chart 2:
            "enroll_hispanic": 7.8,
            "enroll_pacific": 0.1,
            "enroll_black": 2.5,
            "enroll_native": 0.7,
            "enroll_multi_ethnic": 7.1,
```

**Comment by arailsya** [comment]
@jaronheard @hassanshamim 

 If we filter out pre-2009 data, we should be able to avoid the unspecified category. That category is used at the same time that asian and pacific islander students were counted together, so I think for demo day it should be excluded.

null values and zero values are not the same in this data set. nulls are from when the data wasn't collected/reported, and the zeroes are the value 0.

I'm not sure what you mean by indicate what labels to show for what categories. also, those enroll_racegroup values are percents, i think the visual is student counts not % students. if we want counts it will need to be calculated from enroll_current which is the total number of students enrolled at the school.

**Comment by hassanshamim** [comment]
@jaronheard  @arailsya 

http://service.civicpdx.org/neighborhood-development/api/school_demographics_racial_counts

I've added a new endpoint that implements Aurelia's suggestions.  2009 and previous years are excluded so no more `unspecified` category.  Also the `enroll_*` values have been multiplied by the `enroll_current` column to give you the # of each racial group rather than just the percent.

 New endpoint takes the same query params as the original (year, name, type)

**Comment by arailsya** [comment]
@jaronheard I think you are hitting the wrong API endpoint, hassan made a new endpoint that filtered the data for the visualization to exclude some data from the viz: http://service.civicpdx.org/neighborhood-development/api/school_demographics_racial_counts

**Comment by jaronheard** [comment]
#294 implemented, follow-up issue #394 created for @arailsya's comment about API endpoint.
